File: python/binarysearch/leetcode/find_peak_element.py
from typing import List


# 寻找峰值

class Solution:

    # 暴力法，线性查找
    def findPeakElement_1(self, nums: List[int]) -> int:
        for i in range(1, len(nums) - 1):
            if nums[i - 1] < nums[i] > nums[i + 1]:
                return i
        return 0

File: java/algorithm/binarysearch/leetcode/FindPeakElement.java
package algorithm.binarysearch.leetcode;

/**
 * @author roseduan
 * @time 2022/2/10 8:29 上午
 * @description 寻找峰值
 */
public class FindPeakElement {

    /**
     * 直接找最大值，最大值一定是峰值
     */
    public int findPeakElement1(int[] nums) {
        int max = 0;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[max]) {
                max = i;
            }
        }
        return max;
    }
}
